### PR TITLE
ENT-2878: update EnterpriseCustomerCondition to validate course_entitlements

### PR DESCRIPTION
**Pull request by saleem-latif**
**Jira Ticket:** [ENT-2878](https://openedx.atlassian.net/browse/ENT-2878)

__Description:__

This PR adds support for course entitlement in `EnterpriseCustomerCondition`

### CODE REVIEW

File: ecommerce/enterprise/conditions.py [hunk 1]
```diff
@@ -129,6 +130,28 @@ def is_satisfied(self, offer, basket):  # pylint: disable=unused-argument
         username = basket.owner.username
         course_run_ids = []
         for line in basket.all_lines():
+            if line.product.is_course_entitlement_product:
+                try:
+                    response = get_course_info_from_catalog(basket.site, line.product)
+                except (ReqConnectionError, KeyError, SlumberHttpBaseException, Timeout) as exc:
+                    logger.exception(
+                        'Unable to apply enterprise offer because basket contains a course entitlement product'
```
Comment (hasnain-naveed) [inline ecommerce/enterprise/conditions.py]: I think we can add `[Code Redemption Failure]` as a prefix in the log message. Its already been used in the same function.

**Review by hasnain-naveed — APPROVED** [review]
Small suggestion. Otherwise looks fine. 👍

File: ecommerce/coupons/tests/mixins.py [hunk 2]
```diff
@@ -83,6 +84,22 @@ def mock_course_detail_endpoint(self, course, discovery_api_url, course_info=Non
             content_type='application/json'
         )
 
+    def mock_course_detail_endpoint_error(self, course, discovery_api_url, error):
```
Comment (hasnain-naveed) [inline ecommerce/coupons/tests/mixins.py:87]: I have also written this kind of code in my [PR](https://github.com/edx/ecommerce/pull/2981/files#diff-8f14cd685b0a34df81acf481fe1a0817R101). You should merge your PR first, I will rebase mine.

Comment (saleem-latif) [reply ecommerce/coupons/tests/mixins.py:87]: Okay, I will keep an eye out. Once one of the PRs gets merged, the other one can rebase and update accordingly.

File: ecommerce/enterprise/conditions.py [hunk 3]
```diff
@@ -129,6 +130,28 @@ def is_satisfied(self, offer, basket):  # pylint: disable=unused-argument
         username = basket.owner.username
         course_run_ids = []
```
Comment (HammadAhmadWaqas) [inline ecommerce/enterprise/conditions.py]: can we rename this variable or add comment to clear that this variable can contain course_run_id as well as course_key ? generic name could be `course_ids`

**Review by HammadAhmadWaqas — APPROVED** [review]
one comment otherwise looks good to me.
You can use isort/pytint/pycodestyle commands on ecommerce shell to verify code quality before pushing.

**Pull request merged**